packages/app-project/src/shared/components/CollectionsModal/CollectionsModalContainer.js

```diff
@@ -1,125 +1,99 @@
-    }
-  }
+const DEFAULT_HANDLER = () => true
+const defaultNewCollection = {
+  display_name: '',
+  private: false
+}
+
+const CollectionsModalContainer = ({
+  collectionsModalActive = false,
+  setCollectionsModalActive = DEFAULT_HANDLER,
+  subjectID = ''
+}) => {
+  const {
+    addSubjects = DEFAULT_HANDLER,
+    collections = [],
+    createCollection = DEFAULT_HANDLER
+  } = useStore()
+
+  const [newCollection, setNewCollection] = useState(defaultNewCollection)
+  const [selectedCollection, setSelectedCollection] = useState(undefined)
+  const [selectedSubjectID, setSelectedSubjectID] = useState(subjectID)
 
-  addToCollection (event) {
+  const addToCollection = (event) => {
     event.preventDefault()
-    const { selectedCollection, subjectId } = this.state
-    this.props.addSubjects(selectedCollection.id, [subjectId])
+    addSubjects(selectedCollection.id, [selectedSubjectID])
     this.close()
   }
 
-  createCollection (event) {
+  const createNewCollection = (event) => {
     event.preventDefault()
-    const { newCollection, subjectId } = this.state
-    this.props.createCollection(newCollection, [subjectId])
+    createCollection(newCollection, [selectedSubjectID])
     this.close()
   }
 
-  open (subjectId) {
-    this.setState({
-      active: true,
-      subjectId
-    })
+  const onSelect = (event) => {
+    const collection = event.value
+    setSelectedCollection(collection)
   }
 
-  close () {
-    const subjectId = null
-    this.setState({
-      active: false,
-      subjectId
-    })
+  const close = () => {
+    setCollectionsModalActive(false)
+    setSelectedSubjectID(null)
   }
 
-  onSelect (event) {
-    const selectedCollection = event.value
-    this.setState({ selectedCollection })
-  }
-
-  updateCollection (collectionDetails) {
-    this.setState(prevState => {
-      const newCollection = Object.assign(
-        {},
-        prevState.newCollection,
-        collectionDetails
-      )
-      return { newCollection }
-    })
-  }
-
-  render () {
-    const { collections, searchCollections } = this.props
-    const { active, newCollection, selectedCollection } = this.state
-
-    return (
-      <CollectionsModal active={active} closeFn={this.close}>
-        <SelectCollection
-          collections={collections}
-          disabled={!selectedCollection}
-          onSelect={this.onSelect}
-          onSearch={searchCollections}
-          onSubmit={this.addToCollection}
-          selected={selectedCollection}
-        />
-        <CreateCollection
-          disabled={!newCollection.display_name}
-          collection={newCollection}
-          onChange={this.updateCollection}
-          onSubmit={this.createCollection}
-        />
-      </CollectionsModal>
+  const updateCollection = (collectionDetails) => {
+    const prevNewCollectionDetails = newCollection
+    const updatedCollection = Object.assign(
+      {},
+      prevNewCollectionDetails,
+      collectionDetails
     )
```

Comment:
Use spread operators in modern browsers. They're much faster than `Object.assign()`. 

```js
const updatedCollection = {
  ...prevNewCollectionDetails,
  ...collectionDetails
}
```